op-bindings/bindgen/generator_local.go

```diff
@@ -0,0 +1,244 @@
+	contracts, err := generator.readContractsList()
+	if err != nil {
+		return fmt.Errorf("error reading contract list %s: %w", generator.contractsListPath, err)
+	}
+	if len(contracts) == 0 {
+		return fmt.Errorf("no contracts parsed from given contract list: %s", generator.contractsListPath)
+	}
+
+	return generator.processContracts(contracts)
+}
+
+func (generator *bindGenGeneratorLocal) processContracts(contracts []string) error {
+	tempArtifactsDir, err := mkTempArtifactsDir(generator.logger)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := os.RemoveAll(tempArtifactsDir)
+		if err != nil {
+			generator.logger.Error("Error removing temporary artifact directory", "path", tempArtifactsDir, "err", err.Error())
+		} else {
+			generator.logger.Debug("Successfully removed temporary artifact directory")
+		}
+	}()
+
+	sourceMapsList := strings.Split(generator.sourceMapsList, ",")
+	sourceMapsSet := make(map[string]struct{})
+	for _, k := range sourceMapsList {
+		sourceMapsSet[k] = struct{}{}
+	}
+
+	contractArtifactPaths, err := generator.getContractArtifactPaths()
+	if err != nil {
+		return err
+	}
+
+	contractMetadataFileTemplate := template.Must(template.New("localContractMetadata").Parse(localContractMetadataTemplate))
+
+	for _, contractName := range contracts {
+		generator.logger.Info("Generating bindings and metadata for local contract", "contract", contractName)
+
+		forgeArtifact, err := generator.readForgeArtifact(contractName, contractArtifactPaths)
+		if err != nil {
+			return err
+		}
+
+		abiFilePath, bytecodeFilePath, err := writeContractArtifacts(generator.logger, tempArtifactsDir, contractName, forgeArtifact.Abi, []byte(forgeArtifact.Bytecode.Object.String()))
+		if err != nil {
+			return err
+		}
+
+		err = genContractBindings(generator.logger, abiFilePath, bytecodeFilePath, generator.bindingsPackageName, contractName)
+		if err != nil {
+			return err
+		}
+
+		deployedSourceMap, canonicalStorageStr, err := generator.canonicalizeStorageLayout(forgeArtifact, sourceMapsSet, contractName)
+		if err != nil {
+			return err
+		}
+
+		contractMetaData := localContractMetadata{
+			Name:              contractName,
+			StorageLayout:     canonicalStorageStr,
+			DeployedBin:       forgeArtifact.DeployedBytecode.Object.String(),
+			Package:           generator.bindingsPackageName,
+			DeployedSourceMap: deployedSourceMap,
+		}
+
+		if err := generator.writeContractMetadata(contractMetaData, contractName, contractMetadataFileTemplate); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (generator *bindGenGeneratorLocal) getContractArtifactPaths() (map[string]string, error) {
+	// If some contracts have the same name then the path to their
+	// artifact depends on their full import path. Scan over all artifacts
+	// and hold a mapping from the contract name to the contract path.
+	// Walk walks the directory deterministically, so the later instance
```

Comment:
based on the code it looks like the first instance is used?

Also if there's a conflict, worth a warning log?

Reply:
Also adding a warning makes the console output a bit obnoxious

```
WARN [12-03|20:12:45.566] Multiple versions of forge artifacts exist, using lesser version contract=Address
WARN [12-03|20:12:45.566] Multiple versions of forge artifacts exist, using lesser version contract=AddressAliasHelper
WARN [12-03|20:12:45.566] Multiple versions of forge artifacts exist, using lesser version contract=AddressUpgradeable
WARN [12-03|20:12:45.566] Multiple versions of forge artifacts exist, using lesser version contract=AssetReceiver
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=CommonBase
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=ScriptBase
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=TestBase
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=Bytes
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=CallRecorder
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=CallerCaller
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=ConfigurableCaller
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=Reverter
WARN [12-03|20:12:45.567] Multiple versions of forge artifacts exist, using lesser version contract=Chains
WARN [12-03|20:12:45.568] Multiple versions of forge artifacts exist, using lesser version contract=Clone
WARN [12-03|20:12:45.568] Multiple versions of forge artifacts exist, using lesser version contract=ClonesWithImmutableArgs
WARN [12-03|20:12:45.568] Multiple versions of forge artifacts exist, using lesser version contract=CompatibilityFallbackHandler
WARN [12-03|20:12:45.568] Multiple versions of forge artifacts exist, using lesser version contract=DefaultCallbackHandler
WARN [12-03|20:12:45.568] Multiple versions of forge artifacts exist, using lesser version contract=Context
WARN [12-03|20:12:45.568] Multiple versions of forge artifacts exist, using lesser version contract=ContextUpgradeable
WARN [12-03|20:12:45.569] Multiple versions of forge artifacts exist, using lesser version contract=CrossDomainOwnable
WARN [12-03|20:12:45.569] Multiple versions of forge artifacts exist, using lesser version contract=DeployConfig
WARN [12-03|20:12:45.569] Multiple versions of forge artifacts exist, using lesser version contract=Deployer
WARN [12-03|20:12:45.569] Multiple versions of forge artifacts exist, using lesser version contract=DisputeErrors
WARN [12-03|20:12:45.569] Multiple versions of forge artifacts exist, using lesser version contract=DisputeGameFactory
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=GameTypes
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=VMStatuses
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ECDSA
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC1155TokenReceiver
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC20
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC20
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC721
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC721
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC721TokenReceiver
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC721TokenReceiver
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC721TokenReceiver
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=ERC777TokensRecipient
WARN [12-03|20:12:45.570] Multiple versions of forge artifacts exist, using lesser version contract=Encoding
WARN [12-03|20:12:45.571] Multiple versions of forge artifacts exist, using lesser version contract=Enum
WARN [12-03|20:12:45.571] Multiple versions of forge artifacts exist, using lesser version contract=Executables
WARN [12-03|20:12:45.571] Multiple versions of forge artifacts exist, using lesser version contract=Executor
WARN [12-03|20:12:45.571] Multiple versions of forge artifacts exist, using lesser version contract=FFIInterface
WARN [12-03|20:12:45.571] Multiple versions of forge artifacts exist, using lesser version contract=FallbackManager
WARN [12-03|20:12:45.571] Multiple versions of forge artifacts exist, using lesser version contract=BaseGuard
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=Guard
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=GuardManager
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=Hashing
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IBondManager
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IDisputeGame
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IDisputeGameFactory
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IDripCheck
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IERC1271
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IERC165
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IERC721
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IERC721Enumerable
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IFaultDisputeGame
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=Enum
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=Enum
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IGnosisSafe
WARN [12-03|20:12:45.572] Multiple versions of forge artifacts exist, using lesser version contract=IInitializable
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=IMulticall3
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=ILegacyMintableERC20
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=IOptimismMintableERC20
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=IOptimismMintableERC721
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=IOutputBisectionGame
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=IProxyCreationCallback
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=ISemver
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=ISignatureValidator
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=ISignatureValidatorConstants
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=Initializable
WARN [12-03|20:12:45.573] Multiple versions of forge artifacts exist, using lesser version contract=Initializable
WARN [12-03|20:12:45.574] Multiple versions of forge artifacts exist, using lesser version contract=TestERC721
WARN [12-03|20:12:45.574] Multiple versions of forge artifacts exist, using lesser version contract=LegacyCrossDomainUtils
WARN [12-03|20:12:45.574] Multiple versions of forge artifacts exist, using lesser version contract=LibClock
WARN [12-03|20:12:45.574] Multiple versions of forge artifacts exist, using lesser version contract=LibGameId
WARN [12-03|20:12:45.574] Multiple versions of forge artifacts exist, using lesser version contract=LibHashing
WARN [12-03|20:12:45.574] Multiple versions of forge artifacts exist, using lesser version contract=LibPosition
WARN [12-03|20:12:45.574] Multiple versions of forge artifacts exist, using lesser version contract=LibSort
WARN [12-03|20:12:45.575] Multiple versions of forge artifacts exist, using lesser version contract=MerkleTrie
WARN [12-03|20:12:45.575] Multiple versions of forge artifacts exist, using lesser version contract=MockERC20
WARN [12-03|20:12:45.575] Multiple versions of forge artifacts exist, using lesser version contract=IERC721TokenReceiver
WARN [12-03|20:12:45.575] Multiple versions of forge artifacts exist, using lesser version contract=MockERC721
WARN [12-03|20:12:45.575] Multiple versions of forge artifacts exist, using lesser version contract=ModuleManager
WARN [12-03|20:12:45.575] Multiple versions of forge artifacts exist, using lesser version contract=NativeCurrencyPaymentFallback
WARN [12-03|20:12:45.575] Multiple versions of forge artifacts exist, using lesser version contract=IMulticall3
WARN [12-03|20:12:45.576] Multiple versions of forge artifacts exist, using lesser version contract=Ownable
WARN [12-03|20:12:45.576] Multiple versions of forge artifacts exist, using lesser version contract=OwnableUpgradeable
WARN [12-03|20:12:45.576] Multiple versions of forge artifacts exist, using lesser version contract=Owned
WARN [12-03|20:12:45.576] Multiple versions of forge artifacts exist, using lesser version contract=OwnerManager
WARN [12-03|20:12:45.576] Multiple versions of forge artifacts exist, using lesser version contract=PeripheryDeployConfig
WARN [12-03|20:12:45.576] Multiple versions of forge artifacts exist, using lesser version contract=Predeploys
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=RLPReader
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=RLPWriter
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=Safe
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=SafeMath
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=IProxy
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=SafeProxy
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=SafeProxyFactory
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=SafeSigners
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=SafeStorage
WARN [12-03|20:12:45.577] Multiple versions of forge artifacts exist, using lesser version contract=DeployedSafe
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=OwnerSimulator
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SafeTestLib
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SafeTestTools
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=Sort
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=Script
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SecureMerkleTrie
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SecuredTokenTransfer
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SelfAuthorized
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SignMessageLib
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SignatureChecker
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SignatureDecoder
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SimpleStorage
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=SimpleStorage
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=Singleton
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=StdAssertions
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=StdChains
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=StdCheats
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=StdCheatsSafe
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=stdError
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=StdInvariant
WARN [12-03|20:12:45.578] Multiple versions of forge artifacts exist, using lesser version contract=stdJson
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=stdMath
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=stdStorage
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=stdStorageSafe
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=StdStyle
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=StdUtils
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=Storage
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=StorageAccessible
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=Strings
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=TestERC1271Wallet
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=TestERC20
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=TestERC721
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=TestERC721
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=Transactor
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=Types
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=Vm
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=VmSafe
WARN [12-03|20:12:45.579] Multiple versions of forge artifacts exist, using lesser version contract=console
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=console2
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=EIP712
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=IERC165
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=IERC165
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=Types
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=Types
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=safeconsole
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=DSTest
WARN [12-03|20:12:45.580] Multiple versions of forge artifacts exist, using lesser version contract=Test
INFO [12-03|20:12:45.580] Generating bindings and metadata for local contract contract=SystemConfig
INFO [12-03|20:12:45.601] Generating bindings and metadata for local contract contract=L1CrossDomainMessenger
INFO [12-03|20:12:45.618] Generating bindings and metadata for local contract contract=L1StandardBridge
INFO [12-03|20:12:45.638] Generating bindings and metadata for local contract contract=OptimismPortal
INFO [12-03|20:12:45.658] Generating bindings and metadata for local contract contract=L2OutputOracle
INFO [12-03|20:12:45.674] Generating bindings and metadata for local contract contract=AddressManager
INFO [12-03|20:12:45.687] Generating bindings and metadata for local contract contract=L1Block
INFO [12-03|20:12:45.700] Generating bindings and metadata for local contract contract=L2ToL1MessagePasser
INFO [12-03|20:12:45.712] Generating bindings and metadata for local contract contract=GasPriceOracle
INFO [12-03|20:12:45.726] Generating bindings and metadata for local contract contract=L2CrossDomainMessenger
INFO [12-03|20:12:45.741] Generating bindings and metadata for local contract contract=L2StandardBridge
INFO [12-03|20:12:45.759] Generating bindings and metadata for local contract contract=L2ERC721Bridge
INFO [12-03|20:12:45.774] Generating bindings and metadata for local contract contract=L1ERC721Bridge
INFO [12-03|20:12:45.789] Generating bindings and metadata for local contract contract=OptimismMintableERC721Factory
INFO [12-03|20:12:45.803] Generating bindings and metadata for local contract contract=SequencerFeeVault
INFO [12-03|20:12:45.817] Generating bindings and metadata for local contract contract=BaseFeeVault
INFO [12-03|20:12:45.829] Generating bindings and metadata for local contract contract=L1FeeVault
INFO [12-03|20:12:45.842] Generating bindings and metadata for local contract contract=OptimismMintableERC20Factory
INFO [12-03|20:12:45.857] Generating bindings and metadata for local contract contract=OptimismMintableERC20
INFO [12-03|20:12:45.874] Generating bindings and metadata for local contract contract=LegacyERC20ETH
INFO [12-03|20:12:45.891] Generating bindings and metadata for local contract contract=Proxy
INFO [12-03|20:12:45.905] Generating bindings and metadata for local contract contract=ProxyAdmin
INFO [12-03|20:12:45.921] Generating bindings and metadata for local contract contract=LegacyMessagePasser
INFO [12-03|20:12:45.933] Generating bindings and metadata for local contract contract=ERC20
INFO [12-03|20:12:45.949] Generating bindings and metadata for local contract contract=WETH9
INFO [12-03|20:12:45.964] Generating bindings and metadata for local contract contract=DeployerWhitelist
INFO [12-03|20:12:45.978] Generating bindings and metadata for local contract contract=L1BlockNumber
INFO [12-03|20:12:45.990] Generating bindings and metadata for local contract contract=DisputeGameFactory
INFO [12-03|20:12:46.006] Generating bindings and metadata for local contract contract=FaultDisputeGame
INFO [12-03|20:12:46.027] Generating bindings and metadata for local contract contract=OutputBisectionGame
INFO [12-03|20:12:46.047] Generating bindings and metadata for local contract contract=AlphabetVM
INFO [12-03|20:12:46.060] Generating bindings and metadata for local contract contract=StandardBridge
INFO [12-03|20:12:46.077] Generating bindings and metadata for local contract contract=CrossDomainMessenger
INFO [12-03|20:12:46.094] Generating bindings and metadata for local contract contract=MIPS
INFO [12-03|20:12:46.117] Generating bindings and metadata for local contract contract=PreimageOracle
INFO [12-03|20:12:46.130] Generating bindings and metadata for local contract contract=BlockOracle
INFO [12-03|20:12:46.143] Generating bindings and metadata for local contract contract=EAS
INFO [12-03|20:12:46.166] Generating bindings and metadata for local contract contract=SchemaRegistry
INFO [12-03|20:12:46.179] Generating bindings and metadata for local contract contract=ProtocolVersions
INFO [12-03|20:12:46.194] Generating bindings and metadata for local contract contract=Safe
INFO [12-03|20:12:46.218] Generating bindings and metadata for local contract contract=SafeProxyFactory
INFO [12-03|20:12:46.231] Generating bindings and metadata for local contract contract=DelayedVetoable
INFO [12-03|20:12:46.246] Generating bindings and metadata for local contract contract=ISemver
INFO [12-03|20:12:46.258] Generating bindings and metadata for local contract contract=StorageSetter
INFO [12-03|20:12:46.271] Generating bindings and metadata for local contract contract=SuperchainConfig
INFO [12-03|20:12:46.289] Generating bindings and metadata for remote contract contract=MultiCall3
INFO [12-03|20:12:50.206] Generating bindings and metadata for remote contract contract=Create2Deployer
WARN [12-03|20:12:51.014] Unable to compare bytecode from Etherscan against RPC client, no deployment address provided for chain chain=op
INFO [12-03|20:12:51.046] Generating bindings and metadata for remote contract contract=Safe_v130
INFO [12-03|20:12:54.823] Generating bindings and metadata for remote contract contract=SafeL2
INFO [12-03|20:12:57.414] Generating bindings and metadata for remote contract contract=MultiSendCallOnly
INFO [12-03|20:12:58.971] Generating bindings and metadata for remote contract contract=SafeSingletonFactory
INFO [12-03|20:13:00.643] Generating bindings and metadata for remote contract contract=DeterministicDeploymentProxy
INFO [12-03|20:13:02.197] Generating bindings and metadata for remote contract contract=MultiSend
WARN [12-03|20:13:05.158] Unable to compare bytecode from Etherscan against RPC client, no deployment address provided for chain chain=eth
INFO [12-03|20:13:05.388] Generating bindings and metadata for remote contract contract=Permit2
INFO [12-03|20:13:06.638] Generating bindings and metadata for remote contract contract=SenderCreator
INFO [12-03|20:13:07.775] Generating bindings and metadata for remote contract contract=EntryPoint
```